### DISCUSSION TITLE
Breeze build system does not recognize docker-compose version 2.X

### DISCUSSION THREAD

**Discussion by carlsonp** [apache/airflow#19177] [General]
### Apache Airflow version

main (development)

### Operating System

Ubuntu 20.04

### Deployment

Other Docker-based deployment

### Deployment details

When leveraging the `./breeze` script for building from source, the script does not seem to recognize the newest stable release of docker compose version 2.X.  

```
Unknown docker-compose version.  At least 1.29 is needed!  If Breeze fails - upgrade to the latest available docker-compose version
```

Docker-compose has adjusted some of their commands.  `docker-compose version` is now `docker compose version`, note the missing dash.  While there is migration utility script to help in this transition period detailed on their [Github page](https://github.com/docker/compose) it's a separate install.  It might be good to adjust the script to attempt both?

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Are you sure you use the latest version of breeze ? Docker compose 2 support was added few weeks ago and I was using it for quite some time - on Mint not Ubuntu, so i am pretty sure it works.

Can you please update to already main and double check ?

If it still doe snot use for you can you please re-run latest version, set `export VERBOSE_COMMANDS="true"`, re run breeze and paste   results of .build/breeze.out file?

Converting it to discussion now

**Comment by potiuk** [comment]
Maybe you can also paste 'docker-compose --version' output here - it is possible that docker changed the version format (again!)

**Comment by carlsonp** [comment]
```
carlsonp@asus-laptop:~/src/airflow$ git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
```

```
carlsonp@asus-laptop:~/src/airflow$ docker-compose version

Command 'docker-compose' not found, but can be installed with:

sudo snap install docker          # version 20.10.8, or
sudo apt  install docker-compose  # version 1.25.0-1

See 'snap info docker' for additional versions.

carlsonp@asus-laptop:~/src/airflow$ docker compose version
Docker Compose version v2.0.1
```
Here's a partial result of the ending of the verbose `./breeze` output:

```
+ in_container_script_end
+ EXIT_CODE=0
+ [[ 0 != 0 ]]
+ [[ true == \t\r\u\e ]]
+ set +x
+ res=0
+ [[ 0 == \0 ]]
+ rm -f /tmp/tmp.KylaX90upF/out.log
+ verbosity::restore_exit_on_error_status
+ [[ true == \t\r\u\e ]]
+ set -e
+ unset exit_on_error
+ return 0
+ [[ false == \t\r\u\e ]]
+ breeze::run_command /home/carlsonp/src/airflow/.build/dc_ci run --service-ports --rm airflow
+ /home/carlsonp/src/airflow/.build/dc_ci run --service-ports --rm airflow
/home/carlsonp/src/airflow/.build/dc_ci: line 2: docker-compose: command not found

Unknown docker-compose version. At least 1.29 is needed! If Breeze fails - upgrade to latest available docker-compose version

/home/carlsonp/src/airflow/.build/dc_ci: line 68: docker-compose: command not found
+ start_end::script_end
+ local exit_code=127
+ [[ 127 != 0 ]]
+ [[ -f /tmp/tmp.KylaX90upF/out.log ]]
+ start_end::group_end
+ [[ true != \f\a\l\s\e ]]
+ [[ '' == \t\r\u\e ]]
+ echo

+ echo 'ERROR: The previous step completed with error. Please take a look at output above '
ERROR: The previous step completed with error. Please take a look at output above
+ echo

+ [[ false == \t\r\u\e ]]
+ verbosity::print_info '###########################################################################################'
+ [[ false == \t\r\u\e ]]
+ verbosity::print_info '                   EXITING WITH STATUS CODE 127'
+ [[ false == \t\r\u\e ]]
+ verbosity::print_info '###########################################################################################'
+ [[ false == \t\r\u\e ]]
+ [[ true == \t\r\u\e ]]
+ set +x
```

That file doesn't seem to exist?

```
carlsonp@asus-laptop:~/src/airflow$ cat .build/breeze.out
cat: .build/breeze.out: No such file or directory
```
